api/types/events/api.go

```diff
@@ -69,6 +69,12 @@ type AuditEvent interface {
 	GetClusterName() string
 	// SetClusterName sets the name of the teleport cluster on the event.
 	SetClusterName(string)
+
+	// TrimToMaxSize returns a copy of the event trimmed to a smaller
+	// size. The desired size may not be achievable so callers
+	// should always check the size of the returned event before
+	// using it.
```

Comment:
are all the fields in an audit event in scope for trimming? or this is only applied to specific fields?
I think that is worth explaining in this comment.

Reply:
Good idea, added to comment